File: app/src/main/java/org/mifos/selfserviceapp/data/accounts/SavingAccount.java
package org.mifos.selfserviceapp.data.accounts;

/**
 * @author Vishwajeet
 * @since 22/06/16
 */

public class SavingAccount {
    private int id;
    private String accountNo;
    private String productName;

    public String getAccountNo() {
        return accountNo;
    }

    public void setAccountNo(String accountNo) {
        this.accountNo = accountNo;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: app/src/main/java/org/mifos/mobile/models/accounts/savings/SavingAccount.kt
package org.mifos.mobile.models.accounts.savings

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import org.mifos.mobile.models.accounts.Account
import org.mifos.mobile.models.client.DepositType

/**
 * @author Vishwajeet
 * @since 22/06/16
 */

@Parcelize
data class SavingAccount(

        var accountNo: String? = null,

        var productName: String? = null,

        var productId: Int? = null,

        var overdraftLimit: Long = 0,

        var minRequiredBalance: Long = 0,

        var accountBalance: Double = 0.toDouble(),

        var totalDeposits: Double = 0.toDouble(),

        var savingsProductName: String? = null,

        var clientName: String? = null,

        var savingsProductId: String? = null,

        var nominalAnnualInterestRate: Double = 0.toDouble(),

        var status: Status? = null,

        var currency: Currency? = null,

        var depositType: DepositType? = null,

        var lastActiveTransactionDate: List<Int>? = null,

        var timeLine: TimeLine? = null
) : Parcelable, Account() {
    fun isRecurring(): Boolean {
        return this.depositType != null && (this.depositType?.isRecurring() == true)
    }
}